backend/src/main/java/com/bakdata/conquery/sql/conversion/dialect/PostgreSqlFunctionProvider.java

```diff
@@ -13,40 +22,125 @@
  */
 public class PostgreSqlFunctionProvider implements SqlFunctionProvider {
 
+	private static final String INFINITY_DATE_VALUE = "infinity";
+	private static final String MINUS_INFINITY_DATE_VALUE = "-infinity";
+
+	private static final Map<ChronoUnit, DatePart> DATE_CONVERSION = Map.of(
+			ChronoUnit.DECADES, DatePart.DECADE,
+			ChronoUnit.YEARS, DatePart.YEAR,
+			ChronoUnit.DAYS, DatePart.DAY,
+			ChronoUnit.MONTHS, DatePart.MONTH,
+			ChronoUnit.CENTURIES, DatePart.CENTURY
+	);
+
 	@Override
-	public Condition dateRestriction(Field<Object> dateRestrictionColumn, Field<Object> validityDateColumn) {
-		// the && operator checks if two ranges overlap (see https://www.postgresql.org/docs/15/functions-range.html)
-		return DSL.condition(
-				"{0} && {1}",
-				dateRestrictionColumn,
-				validityDateColumn
-		);
+	public Condition dateRestriction(ColumnDateRange dateRestriction, ColumnDateRange validityDate) {
+		if (!validityDate.isSingleColumnRange()) {
+			throw new UnsupportedOperationException("The validity date range has to be converted to a daterange field in the preprocessing step.");
+		}
+		else {
+			// the && operator checks if two ranges overlap (see https://www.postgresql.org/docs/15/functions-range.html)
+			return DSL.condition(
+					"{0} && {1}",
+					dateRestriction.getRange(),
+					validityDate.getRange()
+			);
+		}
 	}
 
 	@Override
-	public Field<Object> daterange(CDateRange dateRestriction) {
-		return DSL.field(
+	public ColumnDateRange daterange(CDateRange dateRestriction) {
+
+		String min;
+		String max;
+
+		if (dateRestriction.isAtLeast()) {
```

Comment:
ihr könnt euch das vereinfachen mit hasLowerBound/hasUpperBound, und dann min/max direkt mit +-Infinity initialisieren.

Wichtig auch zu beachten, es gibt CDateRange.all, also +- Infinity, und die kann in ungünstigen cases auch hier ankommen.

Reply:
Alles klar, danke für den Hinweis! 👍